java/ql/test/library-tests/flow-through-binding/Test.java

```diff
@@ -0,0 +1,42 @@
+public class Test {
+
+  public static Object testFlowThroughSwitchStmt(String s, Integer i, boolean unknown) {
+    Object o = unknown ? s : i;
+    switch (o) {
+      case Integer i2 -> { return i2; }
```

Comment:
`ReturnNode`s are always in the path, so to make sure we're actually testing that the cast is properly detected, we should cast back to `Object` asap.
```suggestion
      case Integer i2 -> { return (Object)i2; }
```